### ISSUE TITLE
Allow speculative trx execution in irreversible mode

### ISSUE THREAD

**Issue by heifner** [AntelopeIO/spring#1030]
With the introduction of Savanna instant finality, there is now no longer a reason to prevent speculative trx execution on irreversible nodes. Allowing speculative execution on irreversible nodes would allow a local node to be used for multiple purposes. 

Particularly exchanges may find this useful. It would allow them to run in irreversible mode, but still submit their trxs to the chain without the need to run a separate node or use a public API.

**Comment by greg7mdp** [comment]
> With the introduction of Savanna instant finality, there is now no longer a reason to prevent speculative trx execution on irreversible nodes

What was the reason pre-Savanna for not executing speculative transactions that Savanna invalidates?

**Comment by heifner** [comment]
> > With the introduction of Savanna instant finality, there is now no longer a reason to prevent speculative trx execution on irreversible nodes
> 
> What was the reason pre-Savanna for not executing speculative transactions that Savanna invalidates?

The state was consider too old, 300+ blocks behind.

**Comment by greg7mdp** [comment]
> The state was consider too old, 300+ blocks behind.

Makes sense, thanks. Even with Savanna, would it not matter that transactions are validated on a state that is two blocks old (or possibly more)?

**Comment by heifner** [comment]
> > The state was consider too old, 300+ blocks behind.
> 
> Makes sense, thanks. Even with Savanna, would it not matter that transactions are validated on a state that is two blocks old (or possibly more)?

You can never speculatively execute on the exact state of the produced block. If the node is only one block away instead of two blocks then that is better, but most use cases it will not make any difference.
